pkg/ice/agent.go

```diff
@@ -126,6 +137,19 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 		portmax:     config.PortMax,
 	}
 
+	// connectionTimeout used to declare a connection dead
+	if config.ConnectionTimeout == 0 {
```

Comment:
I can't think of a case, but could 0 ever be an actual value someone wants? Maybe disabling timeout all together? (Seems unlikely though)

Reply:
That's a good idea, i'll look into that

Reply:
Does it make more sense to create a new null.Duration object, or to just use a pointer and create a copy when the user calls SetConnectionTimeout?

The third option, which i don't like but am including here for completeness, is to use null.Int32 and cast back and forth to time.Duration (it's a typedef underneath it all anyway, but I think this is probably less clear...)

Reply:
I've gone with the second option to
a) keep the AgentConfig struct a bit smaller
b) avoid creating a new null.Duration type

Changing over to use pkg/null is easy if you'd like; this just fits my instinct better.